WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -19,28 +27,23 @@ class CompetitionsController < ApplicationController
   def new
     @js_competitions = @competitions = competitions
     @competition = Competition.new
-
-    render layout: "application"
   end
 
   def index
     @js_competitions = @competitions = competitions
-    render layout: "application"
   end
 
   def create
-    new_competition_params = params.require(:competition).permit(:id, :competition_id_to_clone)
+    new_competition_params = params.require(:competition).permit(:name, :competition_id_to_clone)
     if new_competition_params[:competition_id_to_clone].blank?
       # Creating a blank competition.
       @competition = Competition.new(new_competition_params)
-      # Dummy data to pass validation.
-      @competition.name = @competition.cellName = "Placeholder #{Time.now.year}"
     else
       # Cloning an existing competition!
       competition_to_clone = Competition.find_by_id(new_competition_params[:competition_id_to_clone])
       if competition_to_clone
-        # Don't clone the showAtAll or isConfirmed bits.
-        @competition = Competition.new(competition_to_clone.as_json.merge(new_competition_params).merge(showAtAll: false, isConfirmed: false))
+        # Don't clone showAtAll, isConfirmed, id, name, and cellName.
+        @competition = Competition.new(competition_to_clone.as_json.merge("showAtAll" => false, "isConfirmed" => false, "id" => nil, "name" => nil, "cellName" => nil).merge(new_competition_params))
         @competition.organizers = competition_to_clone.organizers
         @competition.delegates = competition_to_clone.delegates
```

Comment:
I'd move this stuff to `Competition`: `@competition.clone_from(other_competition)`. No need for the controller to know about all the columns a competition table has.